### PR TITLE
fix: bugsnag bug fixes for several destinations and sources

### PR DESCRIPTION
**Pull request by sandeepdsvs**
## Description of the change

> Resolved the following bugsnag errors for 

> Zendesk 
> https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/636ce767cc576a0009584087?filters[event.since]=all&filters[error.status]=open
> https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/636ce63ef739390009b4a54c?filters[event.since]=all&filters[error.status]=open
> 
> Adj
> https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/636908e958dee900088d42e5?filters[event.since]=all&filters[error.status]=open
> 
> Braze
> https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/636c4e7b1fa7df00089d62eb?filters[event.since]=all&filters[error.status]=open
>
> Auth0Source
> https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/636ce7c8a302120008ffe2b1?filters[event.since]=all&filters[error.status]=open
>
> CustomerIo
> https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/636d25f8c423600008f8caea?filters[event.since]=all&filters[error.status]=open
>
> GAEC
> https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/6369b57604dd320008de1674?filters[event.since]=all&filters[error.status]=for_review
> https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/6369d4ec3eb02a0008b11c08?filters[event.since]=all&filters[error.status]=for_review

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/zendesk/transform.js [hunk 1]
```diff
@@ -558,7 +558,6 @@ async function processTrack(message, destinationConfig, headers) {
     userEmail = email;
   }
   zendeskUserID = zendeskUserID || userResponse.data.users[0].id;
-  userEmail = userEmail || userResponse.data.users[0].email;
```
Comment (ItsSudip) [inline v0/destinations/zendesk/transform.js:561]: If the condition in line #545 is satisfied the `userEmail` is not going to get set right?

Comment (sandeepdsvs) [reply v0/destinations/zendesk/transform.js:561]: the `userEmail` is already set and has a validation at lines 535 and 537 respectively. and inside the if condition there is an other check at line 554. So the above line (removed 561 line) is redundant

Comment (sandeepdsvs) [reply v0/destinations/zendesk/transform.js:561]: let me know if I'm wrong

File: v0/destinations/adj/transform.js [hunk 2]
```diff
@@ -18,12 +19,8 @@ const rejectParams = ["revenue", "currency"];
...
   const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
   const { appToken, customMappings, environment } = destination.Config;
-  const platform = message.context.device.type;
-  if (
-    !message.context.device ||
-    !message.context.device.type ||
-    !message.context.device.id
-  ) {
+  const platform = get(message, "context.device.type");
+  if (!platform || !get(message, "context.device.id")) {
```
[the remark this answers is not in the document]
Comment (sandeepdsvs) [reply v0/destinations/adj/transform.js:23]: Is it fine setting message.context?.device?.id to platform? below we're using platform variable to get the device type

File: v0/destinations/braze/transform.js [hunk 3]
```diff
@@ -275,6 +276,7 @@ function processTrackEvent(messageType, message, destination, mappingJson) {
 
   if (
     messageType === EventType.TRACK &&
+    eventName &&
```
[the remark this answers is not in the document]
Comment (sandeepdsvs) [reply v0/destinations/braze/transform.js:279]: updated

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js [hunk 4]
```diff
@@ -61,13 +61,16 @@ const getConversionActionId = async (method, headers, params) => {
     const response = await httpSend(requestBody);
     if (
       !response.success &&
-      !isHttpStatusSuccess(response.response.response.status)
+      !isHttpStatusSuccess(get(response, "response.response.status"))
```
Comment (ItsSudip) [inline v0/destinations/google_adwords_enhanced_conversions/networkHandler.js:64]:
```suggestion
      !isHttpStatusSuccess(response.response?.response?.status)
```

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js [hunk 5]
```diff
@@ -61,13 +61,16 @@ const getConversionActionId = async (method, headers, params) => {
...
     ) {
       throw new ErrorBuilder()
-        .setStatus(response.response.response.status)
-        .setDestinationResponse(response.response.response.data)
+        .setStatus(get(response, "response.response.status"))
+        .setDestinationResponse(get(response, "response.response.data"))
```
Comment (ItsSudip) [inline v0/destinations/google_adwords_enhanced_conversions/networkHandler.js:68]: Same here use optional chaining

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/destinations/adj/transform.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ItsSudip) [reply v0/destinations/adj/transform.js:23]: Got it

File: v0/destinations/zendesk/transform.js (hunk 1 above)
Comment (ItsSudip) [reply v0/destinations/zendesk/transform.js:561]: Got it

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js (hunk 4 above)
Comment (sandeepdsvs) [reply v0/destinations/google_adwords_enhanced_conversions/networkHandler.js]: updated

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js (hunk 5 above)
Comment (sandeepdsvs) [reply v0/destinations/google_adwords_enhanced_conversions/networkHandler.js]: updated

**Review by ItsSudip — APPROVED** [review]

File: v0/destinations/adj/transform.js [hunk 6]
```diff
@@ -18,12 +19,9 @@ const rejectParams = ["revenue", "currency"];
...
   const { appToken, customMappings, environment } = destination.Config;
-  const platform = message.context.device.type;
-  if (
-    !message.context.device ||
-    !message.context.device.type ||
-    !message.context.device.id
-  ) {
+  const platform = get(message, "context.device.type");
+  const id = get(message, "context.device.id");
+  if (!(typeof platform === "string") || !id) {
```
Comment (saikumarrs) [inline v0/destinations/adj/transform.js:24]:
```suggestion
  if (typeof platform !== "string" || !platform || !id) {
```

File: v0/destinations/adj/transform.js [hunk 7]
```diff
@@ -140,6 +139,7 @@ const processRouterDest = async inputs => {
       } catch (error) {
         return getErrorRespEvents(
           [input.metadata],
+          // eslint-disable-next-line no-nested-ternary
```
Comment (saikumarrs) [inline v0/destinations/adj/transform.js:142]: Let's fix the code instead of suppressing the rules.

File: v0/destinations/customerio/transform.js [hunk 8]
```diff
@@ -244,6 +244,10 @@ function responseBuilder(message, evType, evName, destination, messageType) {
           86
         )} Screen`;
       } else {
+        if (!evName) {
+          logger.error(`could not determine eventName`);
+          throw new CustomError(`could not determine eventName`, 400);
```
Comment (saikumarrs) [inline v0/destinations/customerio/transform.js:249]:
```suggestion
          logger.error(`Could not determine event name`);
          throw new CustomError(`Could not determine event name`, 400);
```

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js [hunk 9]
```diff
@@ -133,11 +136,12 @@ const responseHandler = destinationResponse => {
   }
   // else successfully return status, message and original destination response
   const { response } = destinationResponse;
+  const errMessage = get(response, "error.message") || "";
```
Comment (saikumarrs) [inline v0/destinations/google_adwords_enhanced_conversions/networkHandler.js:139]:
```suggestion
  const errMessage = get(response, "error.message", "");
```

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js [hunk 10]
```diff
@@ -133,11 +136,12 @@ const responseHandler = destinationResponse => {
...
   throw new ErrorBuilder()
     .setStatus(status)
     .setDestinationResponse(response)
     .setMessage(
-      `Google_awordds_enhanced_conversion: ${response.error.message} during Google_adwords_enhanced_conversions response transformation`
+      `Google_awordds_enhanced_conversion: ${errMessage} during Google_adwords_enhanced_conversions response transformation`
```
Comment (saikumarrs) [inline v0/destinations/google_adwords_enhanced_conversions/networkHandler.js:144]:
```suggestion
      `Google_awordds_enhanced_conversion: "${errMessage}" during Google_adwords_enhanced_conversions response transformation`
```

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js [hunk 11]
```diff
@@ -61,13 +61,16 @@ const getConversionActionId = async (method, headers, params) => {
...
-        .setStatus(response.response.response.status)
-        .setDestinationResponse(response.response.response.data)
+        .setStatus(response.response?.response?.status)
+        .setDestinationResponse(response.response?.response?.data)
         .setMessage(
-          `Google_awordds_enhanced_conversion: ${response.response.response.data[0].error.message} during Google_adwords_enhanced_conversions response transformation`
+          `Google_awordds_enhanced_conversion: ${get(
+            response,
+            "response.response.data[0].error.message"
+          )} during Google_adwords_enhanced_conversions response transformation`
```
Comment (saikumarrs) [inline v0/destinations/google_adwords_enhanced_conversions/networkHandler.js:73]:
```suggestion
          `Google_awordds_enhanced_conversion: "${get(
            response,
            "response.response.data[0].error.message",
            ""
          )}" during Google_adwords_enhanced_conversions response transformation`
```

Comment (sandeepdsvs) [reply v0/destinations/google_adwords_enhanced_conversions/networkHandler.js]: updated

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js (hunk 10 above)
Comment (sandeepdsvs) [reply v0/destinations/google_adwords_enhanced_conversions/networkHandler.js]: updated

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js (hunk 9 above)
Comment (sandeepdsvs) [reply v0/destinations/google_adwords_enhanced_conversions/networkHandler.js]: updated

File: v0/destinations/customerio/transform.js (hunk 8 above)
Comment (sandeepdsvs) [reply v0/destinations/customerio/transform.js]: updated

**Review by saikumarrs — DISMISSED** [review]

File: __tests__/data/google_adwords_enhanced_conversions_proxy_output.json [hunk 12]
```diff
@@ -2,7 +2,7 @@
   {
     "output": {
       "status": 401,
-      "message": "Google_awordds_enhanced_conversion: Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project. during Google_adwords_enhanced_conversions response transformation",
+      "message": "Google_awordds_enhanced_conversion: \"Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.\" during Google_adwords_enhanced_conversions response transformation",
```
Comment (yashasvibajpai) [inline __tests__/data/google_adwords_enhanced_conversions_proxy_output.json:5]:
```suggestion
      "message": "Google_adwords_enhanced_conversion: \"Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project.\" during Google_adwords_enhanced_conversions response transformation",
```
A small typo

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js [hunk 13]
```diff
@@ -61,13 +61,17 @@ const getConversionActionId = async (method, headers, params) => {
...
+      !isHttpStatusSuccess(response.response?.response?.status)
     ) {
       throw new ErrorBuilder()
-        .setStatus(response.response.response.status)
-        .setDestinationResponse(response.response.response.data)
+        .setStatus(response.response?.response?.status)
+        .setDestinationResponse(response.response?.response?.data)
         .setMessage(
-          `Google_awordds_enhanced_conversion: ${response.response.response.data[0].error.message} during Google_adwords_enhanced_conversions response transformation`
+          `Google_awordds_enhanced_conversion: "${get(
```
Comment (yashasvibajpai) [inline v0/destinations/google_adwords_enhanced_conversions/networkHandler.js:70]:
```suggestion
          `Google_adwords_enhanced_conversion: "${get(
```
Same as above

File: __tests__/data/google_adwords_enhanced_conversions_proxy_output.json (hunk 12 above)
Comment (sandeepdsvs) [reply __tests__/data/google_adwords_enhanced_conversions_proxy_output.json]: Done

File: v0/destinations/google_adwords_enhanced_conversions/networkHandler.js (hunk 13 above)
Comment (sandeepdsvs) [reply v0/destinations/google_adwords_enhanced_conversions/networkHandler.js]: Done

**Review by yashasvibajpai — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**